cql3/query_processor.hh

```diff
@@ -377,6 +377,25 @@ public:
             service::query_state& query_state,
             const std::initializer_list<data_value>& = { });
 
+    future<::shared_ptr<untyped_result_set>> execute_internal(
+            const sstring& query_string,
+            db::consistency_level,
+            const std::vector<std::optional<data_value>>&,
+            cache_internal cache);
+    future<::shared_ptr<untyped_result_set>> execute_internal(
+            const sstring& query_string,
+            db::consistency_level,
+            service::query_state& query_state,
+            const std::vector<std::optional<data_value>>&,
+            cache_internal cache);
+
+    future<::shared_ptr<untyped_result_set>> execute_with_params(
+            statements::prepared_statement::checked_weak_ptr p,
+            db::consistency_level,
+            service::query_state& query_state,
+            const std::vector<std::optional<data_value>>&);
+
+
```

Comment:
Too many new overloads. Let's make the existing ones work. The old data_values_list could have been taught to construct itself from unset markers.

Reply:
Can do.  But I'll base it on `small_vector` based on https://github.com/scylladb/scylladb/pull/16376#discussion_r1438084234